arkouda/pdarraysetops.py

```diff
@@ -495,21 +613,61 @@ def setxor1d(pda1: pdarray, pda2: pdarray,
+    >>> ak.setdiff1d(multia, multib)
+    [array([2, 2, 4, 4, 5, 5]), array([2, 5, 2, 4, 4, 5]), array([2, 4, 5, 4, 2, 5])]
     """
-    if pda1.size == 0:
-        return pda2  # return other pdarray if pda1 is empty
-    if pda2.size == 0:
-        return pda1  # return other pdarray if pda2 is empty
-    if (pda1.dtype == int and pda2.dtype == int) or \
-            (pda1.dtype == akuint64 and pda2.dtype == akuint64):
-        repMsg = generic_msg(cmd="setxor1d", args="{} {} {}". \
-                             format(pda1.name, pda2.name, assume_unique))
-        return create_pdarray(cast(str, repMsg))
-    if not assume_unique:
-        pda1 = cast(pdarray, unique(pda1))
-        pda2 = cast(pdarray, unique(pda2))
-    aux = concatenate((pda1, pda2), ordered=False)
-    aux_sort_indices = argsort(aux)
-    aux = aux[aux_sort_indices]
-    flag = concatenate((array([True]), aux[1:] != aux[:-1], array([True])))
-    return aux[flag[1:] & flag[:-1]]
+    if isinstance(pda1, pdarray) and isinstance(pda2, pdarray):
+        if pda1.size == 0:
+            return pda2  # return other pdarray if pda1 is empty
+        if pda2.size == 0:
+            return pda1  # return other pdarray if pda2 is empty
+        if (pda1.dtype == int and pda2.dtype == int) or \
+                (pda1.dtype == akuint64 and pda2.dtype == akuint64):
+            repMsg = generic_msg(cmd="setxor1d", args="{} {} {}". \
+                                 format(pda1.name, pda2.name, assume_unique))
+            return create_pdarray(cast(str, repMsg))
+        if not assume_unique:
+            pda1 = cast(pdarray, unique(pda1))
+            pda2 = cast(pdarray, unique(pda2))
+        aux = concatenate((pda1, pda2), ordered=False)
+        aux_sort_indices = argsort(aux)
+        aux = aux[aux_sort_indices]
+        flag = concatenate((array([True]), aux[1:] != aux[:-1], array([True])))
+        return aux[flag[1:] & flag[:-1]]
+    elif isinstance(pda1, list) and isinstance(pda2, list):
+        multiarray_setop_validation(pda1, pda2)
+
+        if not assume_unique:
+            ag = GroupBy(pda1)
+            ua = ag.unique_keys
+            bg = GroupBy(pda2)
+            ub = bg.unique_keys
+        else:
+            ua = pda1
+            ub = pda2
+
+        # Key for deinterleaving result
+        isa = concatenate((ones(ua[0].size, dtype=akbool), zeros(ub[0].size, dtype=akbool)), ordered=False)
+        c = [concatenate(x, ordered=False) for x in zip(ua, ub)]
+        g = GroupBy(c)
+        if assume_unique:
+            # need to verify uniqueness, otherwise answer will be wrong
+            if (g.sum(isa)[1] > 1).any():
+                raise ValueError("Called with assume_unique=True, but first argument is not unique")
+            if (g.sum(~isa)[1] > 1).any():
+                raise ValueError("Called with assume_unique=True, but second argument is not unique")
+        k, ct = g.count()
+        truth = g.broadcast(ct == 1, permute=True)
```

Comment:
At this point, instead of broadcasting and re-grouping, you can just do:

```python
single = ct == 1
return [x[single] for x in k]
```

Reply:
Updated to use your suggestion. This was the one I was hoping you'd have a suggestion for. Thanks!

Reply:
Updated to eliminate the gather.